server/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -457,4 +459,7 @@ public long ramBytesUsed() {
     public Set<String> getIndices() {
         return Collections.unmodifiableSet(indices);
     }
+
+    @Override
+    public void close() {}
```

Comment:
`BulkRequest` by virtue of being a transport request already has the ref-counted interface. We should use that and not add `Releasable`, the transport layer uses the ref counted interface on this thing for releasing requests eventually.

Reply:
Not having Releasable will make a lot of unit tests a lot more verbose though -- is that worth it?

Reply:
Yes it's the only way to get this right, we can't have `Releasable` mixed with ref-counting. At least when sending the response via a client that should result in it getting decremented by one, so for those use cases (which I expect to be the majority by a large margin) we shouldn't need any test code adjustment.

Reply:
We had an offline conversation about this. The summary is:

1. We will do explicit ref counting. So sending the response via a client will not result in in getting decremented.
2. We'll avoid using Releasable and try-with-resources as a convenience for calling decRef() in tests
3. We'll add a lot more utility methods and possibly a new RequestBuilder framework (like in #104497) to make the second point not quite as painful.